backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/VerkuendungsImportService.java

```diff
@@ -119,17 +144,74 @@ public VerkuendungImportProcess getStatus(
   }
 
   @Override
-  public void processNormendokumentationspaket(
-    ProcessNormendokumentationspaketUseCase.Query query
-  ) {
-    // TODO: (Malte Laukötter, 2025-04-08) call parseAndValidate with the zip file resource
+  public void processNormendokumentationspaket(ProcessNormendokumentationspaketUseCase.Query query)
+    throws IOException {
+    log.info("Start processing Normendokumentationspaket: {}", query.processId());
+
+    var process = loadVerkuendungImportProcessPort
+      .loadVerkuendungImportProcess(new LoadVerkuendungImportProcessPort.Command(query.processId()))
+      .orElseThrow(() -> new RuntimeException("Could not load verkuendung import process"));
+
+    process =
+    saveVerkuendungImportProcessPort.saveOrUpdateVerkuendungImportProcess(
+      new SaveVerkuendungImportProcessPort.Command(
+        process.getId(),
+        VerkuendungImportProcess.Status.PROCESSING
+      )
+    );
+
+    try {
+      var files = loadNormendokumentationspaketPort.loadNormendokumentationspaket(
+        new LoadNormendokumentationspaketPort.Command(query.processId())
+      );
+      var zipFile = files.file();
+
+      Norm norm = parseAndValidate(zipFile);
+      updateOrSaveNormPort.updateOrSave(new UpdateOrSaveNormPort.Command(norm));
+
+      Verkuendung verkuendung = Verkuendung.builder().eli(norm.getExpressionEli()).build();
+      updateOrSaveVerkuendungPort.updateOrSaveVerkuendung(
+        new UpdateOrSaveVerkuendungPort.Command(verkuendung)
+      );
 
-    throw new RuntimeException("Not implemented yet");
+      saveVerkuendungImportProcessPort.saveOrUpdateVerkuendungImportProcess(
+        new SaveVerkuendungImportProcessPort.Command(
+          process.getId(),
+          VerkuendungImportProcess.Status.SUCCESS
+        )
+      );
+    } catch (Exception e) {
```

Comment:
We catch all exceptions but only create details for "our" exceptions (NormsAppException)? Meaning any other unexpected exception would only be logged but no error detail would be created. Wouldn't this be confusing for the e-V?

Reply:
If there is an error we do not have listed here we do not want to expose the details of it as that could reveal information about our system that could help attackers (see: https://cheatsheetseries.owasp.org/cheatsheets/Error_Handling_Cheat_Sheet.html). And as it is an unexpected exception we should always investigate that anyway and it is probably nothing that the e-V could fix anyway

Reply:
Jep you are completely right. But we should probably put some general info like "internal error"? because otherwise they will get the ERROR state but no info at all